From khali@linux-fr.org Sat Nov 26 19:24:54 2005
Date: Sat, 26 Nov 2005 20:55:35 +0100
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Cc: Laurent Riffard <laurent.riffard@free.fr>
Subject: [PATCH 20/25] i2c: Drop i2c_driver.{owner,name}, 11 of 11
Message-Id: <20051126205535.7a7f6a76.khali@linux-fr.org>
Content-Disposition: inline; filename=i2c-drop-driver-owner-and-name-11-documentation.patch

Document the drop of the owner and name fields of the i2c_driver
structure.

Signed-off-by: Jean Delvare <khali@linux-fr.org>
CC: Laurent Riffard <laurent.riffard@free.fr>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 Documentation/i2c/porting-clients |    4 ++++
 Documentation/i2c/writing-clients |    6 ++++--
 2 files changed, 8 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/Documentation/i2c/porting-clients
+++ gregkh-2.6/Documentation/i2c/porting-clients
@@ -125,6 +125,10 @@ Technical changes:
 
 * [Driver] The flags field of the i2c_driver structure is gone.
   I2C_DF_NOTIFY is now the default behavior.
+  The i2c_driver structure has a driver member, which is itself a
+  structure, those owner and name members should be initialized to
+  THIS_MODULE and a driver name string, respectively. i2c_driver
+  itself has no name member anymore.
 
 Coding policy:
 
--- gregkh-2.6.orig/Documentation/i2c/writing-clients
+++ gregkh-2.6/Documentation/i2c/writing-clients
@@ -25,8 +25,10 @@ routines, a client structure specific in
 address.
 
 static struct i2c_driver foo_driver = {
-	.owner		= THIS_MODULE,
-	.name		= "Foo version 2.3 driver",
+	.driver = {
+		.owner	= THIS_MODULE,
+		.name	= "foo",
+	},
 	.attach_adapter	= &foo_attach_adapter,
 	.detach_client	= &foo_detach_client,
 	.command	= &foo_command /* may be NULL */
